docs/test-fairy/api-reference/upload-api.md

```diff
@@ -8,6 +8,218 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
+     <td><p><small>| OPTIONAL |</small></p><p>Upload file directly to Sauce Labs. Can be "on" or "off". Default is "off".</p></td>
+    </tr>
+  </tbody>
+  <tbody>
+    <tr>
+     <td><code>platform</code></td>
+     <td><p><small>| OPTIONAL |</small></p><p>In case app is not iOS or Android, which are detected automatically, use this to mark an app for specific desktop or console platforms. Values can be "xbox", "playstation", "switch", "windows", "macos". This feature is not enabled by default, please contact support for more information.</p></td>
+    </tr>
+  </tbody>
+</table>
+
+<Tabs
+groupId="params"
+defaultValue="required"
+values={[
+{label: 'Required Params', value: 'required'},
+{label: 'Optional Params', value: 'optional'},
+]}>
+
+<TabItem value="required">
+
+```bash title="Sample Request with Required Params"
+curl https://upload.testfairy.com/api/upload -F api_key='your_api_key' -F file=@sample.apk
+```
+
+</TabItem>
+
+<TabItem value="optional">
+
+```bash title="Sample Request with Optional Params"
+curl https://upload.testfairy.com/api/upload \
+	-F api_key='your_api_key' \
+	-F file=@sample.apk \
+	-F symbols_file=@sample_mapping.txt \
+	-F groups='friends,beta' \
+	-F notify='on' \
+	-F release_notes='stabilitty fixes, improvedment in ui' \
+	-F tags='production, english'
+```
+
+</TabItem>
+</Tabs>
+
+#### Responses
+
+In the case of an error, TestFairy will return a JSON with `status` => `fail` and `code` with one of the values listed below. An additional human-readable error message is supplied to detail the cause of the specific error.
+
+<table id="table-api">
+	<tbody>
+		<tr>
+			<td><code>200</code></td>
+			<td colSpan='2'>Success.</td>
+		</tr>
+	</tbody>
+	<tbody>
+		<tr>
+			<td><code>1</code></td>
+			<td colSpan='2'>Parameter 'xxx' is missing.</td>
+		</tr>
+	</tbody>
+	<tbody>
+		<tr>
+			<td><code>5</code></td>
+			<td colSpan='2'>Invalid API key.</td>
+		</tr>
+	</tbody>
+	<tbody>
+		<tr>
+			<td><code>105</code></td>
+			<td colSpan='2'>Invalid file.</td>
+		</tr>
+	</tbody>
+	
+</table>
+
+```json title="Sample Response"
+{
+	"status": "ok",
+	"app_name": "Jigsaw Puzzlers",
+	"app_version": "0.9.5",
+	"app_url": "https://app.testfairy.com/download/6CWKJCHD60PPVWYJHGM4AADJQYA4SDR0/filename.apk",
+	"landing_page_url": "https://tsfr.io/3tajti",
+}
+```
+
+</details>
+
+### Where can I find my API Key?
+
+In order to get your API KEY, open your account preferences at https://app.testfairy.com/settings/ and click on **Upload API Key**.
+
+### How can I create a new API Key?
+
+In order to create a new API KEY just click on **Regenerate API Key** in your account preferences page.
+
+### Why is my API Key empty?
+
+In cases where we identify that your API KEY was used by mistake to initialize the SDK instead of using your APP TOKEN, we automatically reset the API KEY in order to protect your privacy. In this case, please change the SDK initialization to use the APP TOKEN and create a new API KEY.
```

Comment:
```suggestion
In cases where we identify that your API KEY was used by mistake to initialize the SDK instead of using your APP TOKEN, we automatically reset the API KEY to protect your privacy. In this case, change the SDK initialization to use the APP TOKEN and create a new API KEY.
```